### PR TITLE
Feature/refactored docker task executor

### PR DESCRIPTION
**Pull request by wesleymonte**

### CODE REVIEW

File: src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerCommandExecutor.java [hunk 1]
```diff
@@ -1,38 +1,45 @@
...
+public class DockerCommandExecutor {
+    private HttpWrapper httpWrapper;
+    private final Logger LOGGER = Logger.getLogger(DockerCommandExecutor.class);
     private static final long poolingPeriodTimeMs = 300;
-    private final WorkerDockerRequestHelper workerDockerRequestHelper;
 
-    public DockerCommandExecutor(WorkerDockerRequestHelper workerDockerRequestHelper) {
-        this.workerDockerRequestHelper = workerDockerRequestHelper;
+    public DockerCommandExecutor() {
+        this.httpWrapper = new HttpWrapper();
```
Comment (joivo) [inline src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerCommandExecutor.java]: Is it really necessary to do a `new` in this class? Why not make your methods static?

File: src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerTaskExecutor.java [hunk 2]
```diff
@@ -1,227 +1,95 @@
...
 
 /**
- * This implementation of {@link TaskExecutor} manages the execution of a {@link Task} in a, possible remote,
- * DockerContainer. A new container is created to execute every {@link Task} and destroy after the execution
- * has finished. A task representation is sent to the container, which drives the execution of the
- * commands itself. This objects monitors the execution until it ends on success or failure. If any container
- * initialization error occurs, the return {@link TaskExecutionResult} indicates the Failure.
+ * This implementation of {@link TaskExecutor} manages the execution of a {@link Task} in a,
+ * possible remote, DockerContainer. A new container is created to execute every {@link Task} and
+ * destroy after the execution has finished. A task representation is sent to the container, which
```
Comment (joivo) [inline src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerTaskExecutor.java]: Replace `destroy` by `destroyed`.

File: src/main/java/org/fogbowcloud/arrebol/execution/docker/resource/DockerContainerResource.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package org.fogbowcloud.arrebol.execution.docker.resource;
+
+
+public interface DockerContainerResource {
```
Comment (joivo) [inline src/main/java/org/fogbowcloud/arrebol/execution/docker/resource/DockerContainerResource.java:9]: Add docs in this class.

File: src/main/java/org/fogbowcloud/arrebol/execution/docker/tasklet/TaskletHelper.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
+package org.fogbowcloud.arrebol.execution.docker.tasklet;
+
+import java.util.List;
+import org.apache.log4j.Logger;
+import org.fogbowcloud.arrebol.execution.TaskExecutionResult;
+import org.fogbowcloud.arrebol.execution.docker.DockerCommandExecutor;
+import org.fogbowcloud.arrebol.models.command.Command;
+
+public class TaskletHelper {
+    private static final String taskScriptExecutorFilePath = "/tmp/task-script-executor.sh";
```
Comment (joivo) [inline src/main/java/org/fogbowcloud/arrebol/execution/docker/tasklet/TaskletHelper.java]: Why not follow the pattern of the rest of the constants?

Comment (joivo) [reply src/main/java/org/fogbowcloud/arrebol/execution/docker/tasklet/TaskletHelper.java]: I mean, using only uppercase

File: src/main/java/org/fogbowcloud/arrebol/execution/docker/tasklet/Tasklet.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package org.fogbowcloud.arrebol.execution.docker.tasklet;
+
+import org.fogbowcloud.arrebol.execution.TaskExecutionResult;
+import org.fogbowcloud.arrebol.models.task.Task;
+
+public interface Tasklet {
```
Comment (joivo) [inline src/main/java/org/fogbowcloud/arrebol/execution/docker/tasklet/Tasklet.java:7]: Add docs here too.

**Review by joivo — CHANGES_REQUESTED** [review]

**Comment by joivo** [comment]
Thanks for your effort. Just made a batch of simple comments. Please, verify

File: src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerCommandExecutor.java (hunk 1 above)
Comment (wesleymonte) [reply src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerCommandExecutor.java]: Done.

File: src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerTaskExecutor.java (hunk 2 above)
Comment (wesleymonte) [reply src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerTaskExecutor.java]: Done.

File: src/main/java/org/fogbowcloud/arrebol/execution/docker/tasklet/TaskletHelper.java (hunk 4 above)
Comment (wesleymonte) [reply src/main/java/org/fogbowcloud/arrebol/execution/docker/tasklet/TaskletHelper.java]: Done.

**Review by joivo — APPROVED** [review]

**Pull request merged**